compiler/src/dmd/file_manager.d

```diff
@@ -48,50 +120,59 @@ nothrow:
     *      the found file name or
     *      `null` if it is not different from filename.
     */
-    static const(char)[] lookForSourceFile(const char[] filename, const char*[] path)
+    const(char)[] lookForSourceFile(const char[] filename, const char*[] path)
     {
         //printf("lookForSourceFile(`%.*s`)\n", cast(int)filename.length, filename.ptr);
         /* Search along path[] for .di file, then .d file, then .i file, then .c file.
         */
+        // see if we should check for the module locally.
+        bool checkLocal = packageExists(filename);
```

Comment:
When this is false, it seems all the following code does is needlessly generate filenames with extensions that it won't check. Can't you essentially goto line 176?

Reply:
That’s what I thought too, but they are all used in the foreach loop